src/middlewared/middlewared/plugins/datastore.py

```diff
@@ -211,23 +211,30 @@ def insert(self, name, data, options=None):
         options = options or {}
         prefix = options.get('prefix')
         model = self.__get_model(name)
+        model_field_names = []
         for field in chain(model._meta.fields, model._meta.many_to_many):
             if prefix:
                 name = field.name.replace(prefix, '')
             else:
                 name = field.name
             if name not in data:
                 continue
+            else:
+                model_field_names.append(field.name)
```

Comment:
What's the use case for this? For now I don't think it's a good idea to silently discard unknown field names.

Reply:
I can have this changed, I actually wanted to discuss this if we should be going ahead with this. The reason for introducing this change here was only that when a certificate is being created, we have a lot of other unnecessary values in the data dict. So there are two ways to deal with this, either we modify insert of we clean the data before sending to insert. However it should be noted, in case of updates in datastore, we silently discard unknown field names

Reply:
I think we should

> clean the data before sending to insert

Let's wait for @william-gr 

And I think update should be fixed too

Reply:
I agree its not a good idea to silently discard unknown field names.

If anything we could raise an error if we are sending a attributes that do not exist on the model.

I vote for cleaning the data before calling datastore.